R/data-objects.R

```diff
@@ -11,10 +11,11 @@
 #' @param count Maximum number of bytes to write. Defaults to 2000.
 #' @param truncate Whether to truncate the object when opening it. Defaults to
 #'  `TRUE`.
+#' @param append Whether to append the objects.
+#' @param ticket A valid iRODS ticket string. Defaults to `NULL`.
 #' @param verbose Whether to print information about the HTTP request and
 #'  response. Defaults to `FALSE`.
-#' @param overwrite Whether the file in iRODS should be overwritten if it
-#'  exists. Defaults to `FALSE`.
+#' @param overwrite Deprecated. Use `truncate`.
```

Comment:
How does `truncate` relate to `overwrite`?
And what happens when `append=FALSE`, an overwrite? So the user should make sure first that the data object exists if they want to avoid overwriting?

Reply:
(I see how `truncate` relates in the code below, but it's not clear from this documentation what `truncate` means...

Reply:
Since `count` will be removed from the end-point, `truncate` will have the same effect as `overwrite`. Or do I miss here something. Could also rename `truncate`  to `overwrite`  and deprecate `truncate`.

Reply:
Yes, i think from the user's perspective `overwrite` is more clear (even if under the hood it is sent as `truncate`)
And what happens then with `append=FALSE`?

Reply:
@korydraughn please chime in here.... i think overwrite makes it a bit clearer, as well.

Reply:
`overwrite` should replace `truncate` and `append`.

`truncate` and `append` should not be exposed for a PUT _(assuming that is truly what's being modeled)_.

If you want to expose `truncate` and `append`, I recommend renaming the `iput` function to something that expresses the ability to write data at arbitrary locations within a data object. This also makes it obvious that the call will not trigger PUT specific policy.

> Yes, i think from the user's perspective `overwrite` is more clear (even if under the hood it is sent as `truncate`)
And what happens then with `append=FALSE`?

`truncate` and `append` are independent of each other. Truncation only applies when the data object is first opened. Appending only applies during **write** operations. When append is not enabled, write operations will write data at the location of the file position. When append is enabled, write operations will seek to the end of the file before writing data.

Reply:
Okay, I'll remove both `truncate` and `append` then an only use it internally

Reply:
I have removed the parameters `offset`, `truncate` and `append` from `iput`, `iget`, `isaveRDS` and `ireadRDS`. It will, however, still be allowed as input, but it triggers now a deprecation error. I think this fine level of control was anyway not needed for the targeted users, but maybe I am wrong with that. I considered that offset would similarly not be required when modelling `iput`.